### ISSUE TITLE
Add `BertTokenClassifier` task model and supporting infra

### ISSUE THREAD

**Issue by jbischof** [keras-team/keras-nlp#624]
`BertClassifier` supports one output for the entire input sequence, but per-token classification is important for applications such as POS and NER tagging. This will require some scoping and design:

1. Do we need a separate `BertTokenClassifierPreprocessor` or is just passing a label tensor the same length as the input enough?
2. Do we want to offer a script that can turn a standard dataset like [conll03](https://www.tensorflow.org/datasets/catalog/conll2003) into something our preprocessor can use? In general most token labels are for "words" and not the subword tokens created by WordPiece/SentencePiece.
3. Create a `BertTokenClassifier` task model with the correct task-specific layers and preprocessing.

**Comment by ADITYADAS1999** [comment]
hey [jbischof](https://github.com/jbischof) this issue is still open for contributors?

**Comment by jbischof** [comment]
@ADITYADAS1999 anything tagged with "scoping required" means that we don't know how to do it yet unfortunately! However, if you are super interested in token classification you could prototype it for us in a keras.io example. Let me know if that's interesting to you.

**Comment by ADITYADAS1999** [comment]
> @ADITYADAS1999 anything tagged with "scoping required" means that we don't know how to do it yet unfortunately! However, if you are super interested in token classification you could prototype it for us in a keras.io example. Let me know if that's interesting to you.

thanks for informing [jbischof](https://github.com/jbischof). So this are all ideas but not implemented.

**Comment by jbischof** [comment]
Yes @ADITYADAS1999, anything tagged "scoping required" means we want to do it but we don't know how. We generally save these for ourselves because it's a big ask for contributors to design our API.

**Comment by shivance** [comment]
Hi I've linked a colab with prototype of alignment preprocessor in #754, should help in scoping.
Thoughts @mattdangerw @chenmoneygithub ?
